src/octopeer-github/main/Options/DoNotWatchOptions.ts

```diff
@@ -8,60 +7,34 @@
-/// <reference path="../../content/ElementSelectionBehaviour/SettingElementSelectionBehaviour/AssigneeSettingElementSelectionBehaviour.ts"/>
-/// <reference path="../../content/ElementSelectionBehaviour/SettingElementSelectionBehaviour/LabelSettingElementSelectionBehaviour.ts"/>
-/// <reference path="../../content/ElementSelectionBehaviour/SettingElementSelectionBehaviour/LockConversationSettingElementSelectionBehaviour.ts"/>
-/// <reference path="../../content/ElementSelectionBehaviour/SettingElementSelectionBehaviour/MilestoneSettingElementSelectionBehaviour.ts"/>
-/// <reference path="../../content/ElementSelectionBehaviour/SettingElementSelectionBehaviour/UnsubscribeSettingElementSelectionBehaviour.ts"/>
-/// <reference path="../../content/ElementSelectionBehaviour/TabHeaderElementSelectionBehaviour/CommitsTabHeaderElementSelectionBehaviour.ts"/>
-/// <reference path="../../content/ElementSelectionBehaviour/TabHeaderElementSelectionBehaviour/ConversationTabHeaderElementSelectionBehaviour.ts"/>
-/// <reference path="../../content/ElementSelectionBehaviour/TabHeaderElementSelectionBehaviour/FilesChangedTabHeaderElementSelectionBehaviour.ts"/>
-/* tslint:enable:max-line-length */
+/// <reference path="../../content/ElementSelectionBehaviour/ElementSelectionBehaviourData.ts"/>
+/// <reference path="../../content/ElementSelectionBehaviour/ElementSelectionBehaviourFactory.ts"/>
 
 /**
  * Created by Youri on 19-5-2016.
  * Additional type for storing tuples of Element and Event Creatables.
  */
 type ElementXEventCreatable = {
-    element: ElementSelectionBehaviourCreatable,
+    element: ElementSelectionBehaviourData,
     event: ElementEventBindingCreatable
 };
 
 /**
  * Class for indicating all internal options of the application.
  */
+const factory = new ElementSelectionBehaviourFactory();
 // tslint:disable-next-line:no-unused-variable
 const DoNotWatchOptions = new (class DoNotWatchOptions {
     /**
      * Gets Elements not to Log, from the chrome storage.
      * @returns {ElementSelectionBehaviourCreatable[]}
      */
     public getElements() {
-        let doNotWatchElements: ElementSelectionBehaviourCreatable[] = [];
+        let doNotWatchElements: ElementSelectionBehaviourData[] = [];
         if (Options.getDoNotWatchCommentElements()) {
-            doNotWatchElements.push(CommentInlineCommentButtonElementSelectionBehaviour);
-            doNotWatchElements.push(CommentPRButtonElementSelectionBehaviour);
-            doNotWatchElements.push(EditCommentButtonElementSelectionBehaviour);
+            doNotWatchElements.push(factory.findElementSelectionBehaviourData(ElementID.CONFIRM_INLINE_COMMENT));
+            doNotWatchElements.push(factory.findElementSelectionBehaviourData(ElementID.CREATE_PR_COMMENT));
+            doNotWatchElements.push(factory.findElementSelectionBehaviourData(ElementID.EDIT_COMMENT));
```

Comment:
Just leaving a note here: I will change the `DoNotWatchOptions` class in branch `feedback-contentcontroller-hooktodom`.